src/client.h

```diff
@@ -114,23 +114,43 @@ struct MeshUpdateResult
 	}
 };
 
-class MeshUpdateThread : public UpdateThread
+class MeshUpdateThread : public PoolableUpdateThread<u8>
 {
-private:
-	MeshUpdateQueue m_queue_in;
+public:
+	typedef	u8 MessageType;
 
 protected:
-	virtual void doUpdate();
+	virtual void doUpdate(MessageType &);
```

Comment:
Initially left non-`const` as the `MessageType` instance will be destroyed afterwards and not reused anyway. Will change for const-correctness' sake.